pandas/core/arrays/arrow/array.py

```diff
@@ -553,6 +553,17 @@ def __getitem__(self, item: PositionalIndexer):
             )
         # We are not an array indexer, so maybe e.g. a slice or integer
         # indexer. We dispatch to pyarrow.
+        if isinstance(item, slice):
+            if item.start == item.stop:
+                pass
+            elif (
+                item.stop is not None
+                and item.stop < -len(self)
+                and item.step is not None
+                and item.step < 0
+            ):
```

Comment:
Can you add a comment about why we do this (or refer to a pyarrow bug report or this PR)

Reply:
added the arrow ref